test/extended/apiserver/graceful_termination.go

```diff
@@ -61,6 +62,45 @@ var _ = g.Describe("[sig-api-machinery][Feature:APIServer][Late]", func() {
 		}
 	})
 
+	g.It("kubelet terminates kube-apiserver gracefully extended", func() {
+		controlPlaneTopology, err := exutil.GetControlPlaneTopology(oc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		if *controlPlaneTopology == configv1.ExternalTopologyMode {
+			g.Skip("oc must-gather doesn't currently support external controlPlaneTopology")
+		}
+
+		tempDir, err := ioutil.TempDir("", "test.oc-adm-must-gather.")
+		o.Expect(err).NotTo(o.HaveOccurred())
+		defer os.RemoveAll(tempDir)
+
+		var finalMessageBuilder strings.Builder
+		terminationRegexp := regexp.MustCompile(`Previous pod .* did not terminate gracefully`)
```

Comment:
Why doesn't the  kube-apiserver itself produce an event when this is detected?

Reply:
It [does](https://github.com/openshift/kubernetes/blob/6892b57d65d25fb0588693bce3d338d8b8b1d2b4/cmd/watch-termination/main.go#L149), we even have a test for [it](https://github.com/openshift/origin/blob/master/test/extended/apiserver/graceful_termination.go#L53). The issue is that sometimes the events are not reported (probably deserves more investigation). For example, the test passed in [the faulty run](https://prow.ci.openshift.org/view/gs/test-platform-results/logs/periodic-ci-openshift-release-master-ci-4.18-e2e-gcp-ovn-upgrade/1822964770265894912) even though the servers were not gracefully terminated.